File: 1487-making-file-names-unique/1487-making-file-names-unique.js
/**
 * @param {string[]} names
 * @return {string[]}
 */
var getFolderNames = function(names) {
  const set = new Set();
  
  for (let name of names) {
    if (set.has(name)) {
      let count = 1;
      while (set.has(`${name}(${count})`)) count++;
      
      set.add(`${name}(${count})`);
    }
    
    set.add(name);
  }
  
  return [...set];
};

File: 1487-making-file-names-unique/1487-making-file-names-unique.py
class Solution:
    def getFolderNames(self, names: List[str]) -> List[str]:
        d = {}
        for i, name in enumerate(names):
            if name in d:
                k = d[name]
                while f'{name}({k})' in d:
                    k += 1
                d[f'{name}({k})'] = 1
                d[name] = k + 1
                names[i] = f'{name}({k})'
            d[name] = 1
        return names
        
        
        
